### PR TITLE
python3Packages.threadpoolctl: 2.1.0 -> 2.2.0

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Motivation for this change

Routine update.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [NA] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133074` at 4ce82097 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.cnvkit</li>
    <li>python38Packages.dask-ml</li>
    <li>python38Packages.dask-xgboost</li>
    <li>python38Packages.ignite</li>
    <li>python38Packages.textacy</li>
    <li>python39Packages.cnvkit</li>
    <li>python39Packages.dask-ml</li>
    <li>python39Packages.dask-xgboost</li>
    <li>python39Packages.ignite</li>
    <li>python39Packages.textacy</li>
  </ul>
</details>
<details>
  <summary>17 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/2f31d39d68645314e1964ede6e84c067">python38Packages.hmmlearn</a></li>
    <li><a href="https://gist.github.com/491a5bf22385d6848aab740f91b7e744">python38Packages.imbalanced-learn</a></li>
    <li><a href="https://gist.github.com/ee74460468395e755147be7932c3236d">python38Packages.kmapper</a></li>
    <li><a href="https://gist.github.com/d0f9ef6ed6268befa8ee65a517729c13">python38Packages.mlxtend</a></li>
    <li><a href="https://gist.github.com/915fd2894de1c23e73ca60a51833022c">python38Packages.pygbm</a></li>
    <li><a href="https://gist.github.com/172fb2f0830a615df20480940fddcdeb">python38Packages.sklearn-deap</a></li>
    <li><a href="https://gist.github.com/13ec4c7cb3693064afb92bbc957508bc">python39Packages.hmmlearn</a></li>
    <li><a href="https://gist.github.com/36978f94247d743a6e28cb4a493ea1af">python39Packages.imbalanced-learn</a></li>
    <li><a href="https://gist.github.com/7fd7554671ce29eb16fd2e6376bebfda">python39Packages.kmapper</a></li>
    <li><a href="https://gist.github.com/a5fc34d696cd79b6c558a7180ddc6424">python39Packages.mlxtend</a></li>
    <li><a href="https://gist.github.com/a683f66720a3970cd2e55c2fb823796f">python39Packages.pygbm</a></li>
    <li><a href="https://gist.github.com/e473dff82854348561a48e4c326c6653">python39Packages.qiskit</a></li>
    <li><a href="https://gist.github.com/4eb418175a6ef3bd1786ea299a935d06">python39Packages.qiskit-aqua</a></li>
    <li><a href="https://gist.github.com/3c69d844f49d9a8057f218bfea2bd5a0">python39Packages.qiskit-ignis</a></li>
    <li><a href="https://gist.github.com/821d57052b1ef9266e5d0eafb3e8d488">python39Packages.scikit-tda</a></li>
    <li><a href="https://gist.github.com/a69da419237b4e7821237044542f86f5">python39Packages.sklearn-deap</a></li>
    <li><a href="https://gist.github.com/c204afb47610f439540cf339fa460168">python39Packages.umap-learn</a></li>
  </ul>
</details>
<details>
  <summary>28 packages skipped due to time constraints:</summary>
  <ul>
    <li>python38Packages.dask-glm</li>
    <li>python38Packages.graspologic</li>
    <li>python38Packages.nilearn</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.osmnx</li>
    <li>python38Packages.pytorch-metric-learning</li>
    <li>python38Packages.qiskit</li>
    <li>python38Packages.qiskit-aqua</li>
    <li>python38Packages.qiskit-ignis</li>
    <li>python38Packages.rising</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>32 packages built successfully:</summary>
  <ul>
    <li>python38Packages.batchgenerators</li>
    <li>python38Packages.bayesian-optimization</li>
    <li>python38Packages.hdbscan</li>
    <li>python38Packages.hyppo</li>
    <li>python38Packages.librosa</li>
    <li>python38Packages.lightgbm</li>
    <li>python38Packages.mlrose</li>
    <li>python38Packages.persim</li>
    <li>python38Packages.ppscore</li>
    <li>python38Packages.pynndescent</li>
    <li>python38Packages.ripser</li>
    <li>python38Packages.scikit-learn</li>
    <li>python38Packages.seqeval</li>
    <li>python38Packages.threadpoolctl</li>
    <li>python38Packages.vowpalwabbit</li>
    <li>python38Packages.word2vec</li>
    <li>python39Packages.batchgenerators</li>
    <li>python39Packages.bayesian-optimization</li>
    <li>python39Packages.hdbscan</li>
    <li>python39Packages.hyppo</li>
    <li>python39Packages.librosa</li>
    <li>python39Packages.lightgbm</li>
    <li>python39Packages.mlrose</li>
    <li>python39Packages.persim</li>
    <li>python39Packages.ppscore</li>
    <li>python39Packages.pynndescent</li>
    <li>python39Packages.ripser</li>
    <li>python39Packages.scikit-learn</li>
    <li>python39Packages.seqeval</li>
    <li>python39Packages.threadpoolctl</li>
    <li>python39Packages.vowpalwabbit</li>
    <li>python39Packages.word2vec</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `flit`.
Near pkgs/development/python-modules/threadpoolctl/default.nix:5:3:
```
  |
5 | , flit
  |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/python-modules/threadpoolctl/default.nix [hunk 1]
```diff
@@ -20,7 +20,7 @@ buildPythonPackage rec {
     owner = "joblib";
     repo = pname;
     rev = version;
-    sha256 = "0sl6mp3b2gb0dvqkhnkmrp2g3r5c7clyyyxzq44xih6sw1pgx9df";
+    sha256 = "7UUjbX1IpXtUAgN48Db43Zr1u360UETSUnIHD6rQRLs=";
   };
 
   checkInputs = [ pytestCheckHook pytest-cov numpy scipy ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/threadpoolctl/default.nix:26]: Can we patch out the usage of pytest-cov?

Comment (jonringer) [reply pkgs/development/python-modules/threadpoolctl/default.nix:26]: it should be optional if we use pytestCheckHook

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 133074` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
14 packages marked as broken and skipped:
</summary>
<ul>
<li>python38Packages.dask-xgboost</li>
<li>python38Packages.mlxtend</li>
<li>python38Packages.optuna</li>
<li>python38Packages.pygbm</li>
<li>python38Packages.shap</li>
<li>python38Packages.sklearn-deap</li>
<li>python38Packages.skorch</li>
<li>python39Packages.dask-xgboost</li>
<li>python39Packages.mlxtend</li>
<li>python39Packages.optuna</li>
<li>python39Packages.pygbm</li>
<li>python39Packages.shap</li>
<li>python39Packages.sklearn-deap</li>
<li>python39Packages.skorch</li>
</ul>
</details>
<details>

  <summary>8 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python38Packages.graspologic: log was empty</li>
<li>python38Packages.pytorch-metric-learning: log was empty</li>
<li>python38Packages.scikit-optimize: log was empty</li>
<li>python38Packages.textacy: log was empty</li>
<li>python39Packages.graspologic: log was empty</li>
<li>python39Packages.pytorch-metric-learning: log was empty</li>
<li>python39Packages.scikit-optimize: log was empty</li>
<li>python39Packages.textacy: log was empty</li>
  </ul>
</details>
<details>
<summary>
68 packages built:
</summary>
<ul>
<li>python38Packages.batchgenerators</li>
<li>python38Packages.bayesian-optimization</li>
<li>python38Packages.cnvkit</li>
<li>python38Packages.dask-glm</li>
<li>python38Packages.dask-ml</li>
<li>python38Packages.hdbscan</li>
<li>python38Packages.hmmlearn</li>
<li>python38Packages.hyppo</li>
<li>python38Packages.ignite</li>
<li>python38Packages.imbalanced-learn</li>
<li>python38Packages.kmapper</li>
<li>python38Packages.librosa</li>
<li>python38Packages.lightgbm</li>
<li>python38Packages.mlrose</li>
<li>python38Packages.nilearn</li>
<li>python38Packages.osmnx</li>
<li>python38Packages.persim</li>
<li>python38Packages.ppscore</li>
<li>python38Packages.pynndescent</li>
<li>python38Packages.qiskit</li>
<li>python38Packages.qiskit-aqua</li>
<li>python38Packages.qiskit-ignis</li>
<li>python38Packages.ripser</li>
<li>python38Packages.rising</li>
<li>python38Packages.scikit-bio</li>
<li>python38Packages.scikit-learn</li>
<li>python38Packages.scikit-tda</li>
<li>python38Packages.seqeval</li>
<li>python38Packages.threadpoolctl</li>
<li>python38Packages.umap-learn</li>
<li>python38Packages.vowpalwabbit</li>
<li>python38Packages.word2vec</li>
<li>python38Packages.xgboost</li>
<li>python39Packages.batchgenerators</li>
<li>python39Packages.bayesian-optimization</li>
<li>python39Packages.cnvkit</li>
<li>python39Packages.dask-glm</li>
<li>python39Packages.dask-ml</li>
<li>python39Packages.hdbscan</li>
<li>python39Packages.hmmlearn</li>
<li>python39Packages.hyppo</li>
<li>python39Packages.ignite</li>
<li>python39Packages.imbalanced-learn</li>
<li>python39Packages.kmapper</li>
<li>python39Packages.librosa</li>
<li>python39Packages.lightgbm</li>
<li>python39Packages.mlrose</li>
<li>python39Packages.nilearn</li>
<li>python39Packages.osmnx</li>
<li>python39Packages.persim</li>
<li>python39Packages.ppscore</li>
<li>python39Packages.pynndescent</li>
<li>python39Packages.qiskit</li>
<li>python39Packages.qiskit-aqua</li>
<li>python39Packages.qiskit-ignis</li>
<li>python39Packages.ripser</li>
<li>python39Packages.rising</li>
<li>python39Packages.scikit-bio</li>
<li>python39Packages.scikit-learn</li>
<li>python39Packages.scikit-tda</li>
<li>python39Packages.seqeval</li>
<li>python39Packages.threadpoolctl</li>
<li>python39Packages.umap-learn</li>
<li>python39Packages.vowpalwabbit</li>
<li>python39Packages.word2vec</li>
<li>python39Packages.xgboost</li>
<li>tebreak</li>
<li>tts</li>
</ul>
</details>

File: pkgs/development/python-modules/threadpoolctl/default.nix (hunk 1 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/threadpoolctl/default.nix]: Indeed, removed, thanks.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 133074` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
14 packages marked as broken and skipped:
</summary>
<ul>
<li>python38Packages.dask-xgboost</li>
<li>python38Packages.mlxtend</li>
<li>python38Packages.optuna</li>
<li>python38Packages.pygbm</li>
<li>python38Packages.shap</li>
<li>python38Packages.sklearn-deap</li>
<li>python38Packages.skorch</li>
<li>python39Packages.dask-xgboost</li>
<li>python39Packages.mlxtend</li>
<li>python39Packages.optuna</li>
<li>python39Packages.pygbm</li>
<li>python39Packages.shap</li>
<li>python39Packages.sklearn-deap</li>
<li>python39Packages.skorch</li>
</ul>
</details>
<details>

  <summary>8 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python38Packages.graspologic: log was empty</li>
<li>python38Packages.pytorch-metric-learning: log was empty</li>
<li>python38Packages.scikit-optimize: log was empty</li>
<li>python38Packages.textacy: log was empty</li>
<li>python39Packages.graspologic: log was empty</li>
<li>python39Packages.pytorch-metric-learning: log was empty</li>
<li>python39Packages.scikit-optimize: log was empty</li>
<li>python39Packages.textacy: log was empty</li>
  </ul>
</details>
<details>
<summary>
68 packages built:
</summary>
<ul>
<li>python38Packages.batchgenerators</li>
<li>python38Packages.bayesian-optimization</li>
<li>python38Packages.cnvkit</li>
<li>python38Packages.dask-glm</li>
<li>python38Packages.dask-ml</li>
<li>python38Packages.hdbscan</li>
<li>python38Packages.hmmlearn</li>
<li>python38Packages.hyppo</li>
<li>python38Packages.ignite</li>
<li>python38Packages.imbalanced-learn</li>
<li>python38Packages.kmapper</li>
<li>python38Packages.librosa</li>
<li>python38Packages.lightgbm</li>
<li>python38Packages.mlrose</li>
<li>python38Packages.nilearn</li>
<li>python38Packages.osmnx</li>
<li>python38Packages.persim</li>
<li>python38Packages.ppscore</li>
<li>python38Packages.pynndescent</li>
<li>python38Packages.qiskit</li>
<li>python38Packages.qiskit-aqua</li>
<li>python38Packages.qiskit-ignis</li>
<li>python38Packages.ripser</li>
<li>python38Packages.rising</li>
<li>python38Packages.scikit-bio</li>
<li>python38Packages.scikit-learn</li>
<li>python38Packages.scikit-tda</li>
<li>python38Packages.seqeval</li>
<li>python38Packages.threadpoolctl</li>
<li>python38Packages.umap-learn</li>
<li>python38Packages.vowpalwabbit</li>
<li>python38Packages.word2vec</li>
<li>python38Packages.xgboost</li>
<li>python39Packages.batchgenerators</li>
<li>python39Packages.bayesian-optimization</li>
<li>python39Packages.cnvkit</li>
<li>python39Packages.dask-glm</li>
<li>python39Packages.dask-ml</li>
<li>python39Packages.hdbscan</li>
<li>python39Packages.hmmlearn</li>
<li>python39Packages.hyppo</li>
<li>python39Packages.ignite</li>
<li>python39Packages.imbalanced-learn</li>
<li>python39Packages.kmapper</li>
<li>python39Packages.librosa</li>
<li>python39Packages.lightgbm</li>
<li>python39Packages.mlrose</li>
<li>python39Packages.nilearn</li>
<li>python39Packages.osmnx</li>
<li>python39Packages.persim</li>
<li>python39Packages.ppscore</li>
<li>python39Packages.pynndescent</li>
<li>python39Packages.qiskit</li>
<li>python39Packages.qiskit-aqua</li>
<li>python39Packages.qiskit-ignis</li>
<li>python39Packages.ripser</li>
<li>python39Packages.rising</li>
<li>python39Packages.scikit-bio</li>
<li>python39Packages.scikit-learn</li>
<li>python39Packages.scikit-tda</li>
<li>python39Packages.seqeval</li>
<li>python39Packages.threadpoolctl</li>
<li>python39Packages.umap-learn</li>
<li>python39Packages.vowpalwabbit</li>
<li>python39Packages.word2vec</li>
<li>python39Packages.xgboost</li>
<li>tebreak</li>
<li>tts</li>
</ul>
</details>

**Comment by risicle** [comment]
On macos 10.15 I'm now getting:

```
builder for '/nix/store/z0fckxwqffxwfddckhzv0vav76zwm9kd-python3.8-threadpoolctl-2.2.0.drv' failed with exit code 1; last 10 log lines:
          )
          for module in threadpool_info():
              if module["internal_api"] == "openblas":
  >               assert module["architecture"] in expected_openblas_architectures
  E               AssertionError: assert 'Core2' in ('armv8', 'Haswell', 'SkylakeX', 'Sandybridge')

  tests/test_threadpoolctl.py:460: AssertionError
  =========================== short test summary info ============================
  FAILED tests/test_threadpoolctl.py::test_architecture - AssertionError: asser...
  =================== 1 failed, 18 passed, 31 skipped in 2.36s ===================
```

**Pull request merged**